File: cs365/hw3/Queue/ArrayQueue.java
// ArrayQueue.java
// Mitchell Wade
// Feb. 3, 2015
//
// This program implements the Queue.java interface.

import java.util.ArrayList;

class ArrayQueue implements Queue {
    
    ArrayList<Object> list = new ArrayList<Object>();
    
    public void add(Object o) {
        list.add(o);
    }
    public Object remove() {
        return list.remove(0);
    }
    public boolean isEmpty() {
        return list.isEmpty();
    }
}

File: cs365/hw4/Queue/ArrayQueue.cpp
// ArrayQueue.cpp
// Mitchell Wade
// Feb. 9, 2015

#include "Queue.h"
#include <list>

using namespace std;

// Add a value to the queue
void ArrayQueue::add(int value) {
    
    Q.push_back(value);
}

// Remove a value off the queue
int ArrayQueue::remove() {
    
    int rv;    
    
    rv = Q.front();
    Q.pop_front();

    return rv;
}

// Return if it is empty or not
bool ArrayQueue::isEmpty() {
    
    return Q.empty();
}

